src/registry/check.rs

```diff
@@ -80,100 +60,19 @@ pub(crate) fn command(
         None
     };
 
-    // Load the semantic convention registry into a local registry repo.
-    // No parsing errors should be observed.
-    let main_semconv_specs = load_semconv_specs(
-        &main_registry_repo,
-        logger.clone(),
-        args.common_registry_args.follow_symlinks,
-    )
-    .capture_non_fatal_errors(&mut diag_msgs)?;
     let baseline_semconv_specs = baseline_registry_repo
         .as_ref()
         .map(|repo| {
             // Baseline registry resolution should allow non-future features
             // and warnings against it should be suppressed when evaluating
             // against it as a "baseline".
-            load_semconv_specs(
-                repo,
-                logger.clone(),
-                args.common_registry_args.follow_symlinks,
-            )
-            .ignore(|e| matches!(e.severity(), Some(miette::Severity::Warning)))
-            .capture_non_fatal_errors(&mut diag_msgs)
+            load_semconv_specs(repo, logger.clone(), args.registry.follow_symlinks)
+                .ignore(|e| matches!(e.severity(), Some(miette::Severity::Warning)))
```

Comment:
I believe this should actually fail when `--future` is enabled.  I forget how we handled `--future`, so @lquerel can confirm/answer questions if you need, just want to make sure that case is handled.

Reply:
The current behaviour for the baseline registry (unchanged by this refactor) is to ignore all warnings, this includes those that would be promoted to Errors with `--future`. To me this makes sense for the baseline registry since it's likely older and therefore may now fail newer validations.

Reply:
Ah, I missed this was loading the baseline.  Thanks for clarifying!